pangeo_forge_recipes/injections.py

```diff
@@ -1,5 +1,9 @@
 def get_injection_specs():
     return {
+        "Config": {
+            "target_storage": "TARGET_STORAGE",
+            "input_cache_storage": "INPUT_CACHE_STORAGE",
+        },
         "StoreToZarr": {
             "target_root": "TARGET_STORAGE",
         },
```

Comment:
```suggestion
```

Should we take this a step further, and actually use this config object internally?

Meaning delete the StoreToZarr, WriteCombinedReference, and OpenURLWithFSSpec injections specs here, and instead doing something like:


```diff
# transforms.py

from .config import Config

c = Config()
...

class StoreToZarr(PTransform):
    ...
-   target_root: Union[str, FSSpecTarget, RequiredAtRuntimeDefault] = field(
-       default_factory=RequiredAtRuntimeDefault
+   target_root: Union[str, FSSpecTarget, RequiredAtRuntimeDefault] = c.target_storage
```

Without fiddling with it a bit I'm not sure exactly how this would work, but conceptually, what do we think about making `Config` the interface to all injections, including for internal transforms?

Reply:
Given that dependency injection is targeting the recipe this won't work AFAICT without more complicated acrobatics in the AST rewriter which I think we should avoid. So my feedback here is that we just do what you are proposing in the recipes explicitly:

### benefits
* recipes < `0.10.x` (some version with this change) can easily be backward compatible and still leverage the existing dep injection

* recipes > `0.10.x` (some version with this change) are now explicit and folks need less explanation about how this all works (except the explanation about dep injection). For example, with this change you probably don't need to even write documentation about how to write custom transforms b/c `StoreToZarr`, `WriteCombinedReference` and `OpenURLWithFSSpec` recipe examples already show it how it works

* it's only a couple/few call sites we'd be passing `conf.target_storage` and `conf.input_cache_storage` to per recipe

### drawbacks

* if the number of transforms we write requiring `conf.target_storage` and `conf.input_cache_storage` dramatically increases then, yes, it'd seem we might be doing an unnecessary amount of input passing